backend/product-listing.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+	"path"
+	"strings"
+)
+
+const (
+	SAMPLE_TEMPLATE_FOLDER = "samples_templates"
+)
+
+type ProductPage struct {
+	Products     []Product
+	SearchAction string
+}
+
+type Product struct {
+	Name  string
+	Image string
+	URL   string
+	Stars string
+}
+
+var products []Product
+
+func InitProductListing() {
+	products = make([]Product, 0)
+	products = append(products,
+		Product{"Nexus 5x",
+			"https://lh3.googleusercontent.com/O00xttgXZHWYdpiWi7K0y7J2kvBMRXwlumBQNLV1853AMtf7MBBMSJ_ug9MDyUxMNjyo",
```

Comment:
I don't understand this. The product sample has only one product which is a Nexus 5. What's the point of pointing all of those products to the same one?